### ISSUE TITLE
Exception thrown by argument constraint factory is wrapped in TargetInvocationException, not UserCallbackException

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#1646]

**Comment by thomaslevesque** [comment]
@blairconrad have you started working on this already? I hadn't noticed you had assigned it to yourself, and I just implemented a fix...

**Comment by thomaslevesque** [comment]
Well, I *almost* implemented a fix... I think the error message needs some work!

```
FluentAssertions.Execution.AssertionFailedException : Expected string to be 
"Argument constraint expression <constraintFactory()> threw an exception. See inner exception for details." with a length of 105, but 
"Argument constraint expression <Invoke(value(FakeItEasy.Specs.UserCallbackExceptionSpecs+<>c__DisplayClass2_0).constraintFactory)> threw an exception. See inner exception for details." has a length of 183.
```

**Comment by thomaslevesque** [comment]
OK, that works better if the constraint factory is a method rather than a delegate.
Do you think we should try to "clean up" the expression to have `constraintFactory()` in the message instead of `Invoke(value(FakeItEasy.Specs.UserCallbackExceptionSpecs+<>c__DisplayClass2_0).constraintFactory)` when a delegate is used?

**Comment by blairconrad** [comment]
Hi! No, I'm not working on it. Feel free.

I hadn't given much thought to the error message.
We don't provide a huge amount of detail in most cases where we throw a UserCallbackException:
![image](https://user-images.githubusercontent.com/3275797/66261442-51cf5480-e79b-11e9-8a05-c5b254d35531.png)

If cleaning up the expression is at all tricky, I'd maybe just use a semi-generic message like those examples.  The inner exception will provide additional context if need, no?

**Comment by thomaslevesque** [comment]
> No, I'm not working on it. Feel free.

Cool! I pushed my branch.

> We don't provide a huge amount of detail in most cases where we throw a UserCallbackException:

No, but when we can provide a representation of the callback that threw, we do. In this case it's an expression, so we can just `ToString()` it.

> If cleaning up the expression is at all tricky, I'd maybe just use a semi-generic message like those examples. The inner exception will provide additional context if needed, no?

The inner exception does provide details, but it doesn't say what the faulty expression was. I'm not going to try to improve the expression right now. The problem exists in other places as well, and if we want to do something about it, I think we should do it separately.

**Comment by blairconrad** [comment]
This change has been released as part of [FakeItEasy 5.3.0](https://github.com/FakeItEasy/FakeItEasy/releases/tag/5.3.0).
